vulnfeeds/cmd/download_cves/main.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"bufio"
+	"compress/gzip"
+	"crypto/sha256"
+	"encoding/hex"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	cveURLBase   = "https://nvd.nist.gov/feeds/json/cve/1.1/"
+	fileNameBase = "nvdcve-1.1-"
+	startingYear = 2002
+	cvePath      = "cve_jsons"
+)
+
+func main() {
+	currentYear := time.Now().Year()
+	for i := startingYear; i <= currentYear; i++ {
+		downloadCVEAsNeeded(strconv.Itoa(i))
+	}
+	downloadCVEAsNeeded("modified")
+	downloadCVEAsNeeded("recent")
+}
+
+func downloadCVEAsNeeded(version string) {
+	file, err := os.OpenFile(cvePath+"/"+fileNameBase+version+".json", os.O_CREATE|os.O_RDWR, 0644)
+	defer file.Close()
+	if err != nil { // There's an existing file, check if it matches server file
+		log.Fatalf("Something's went wrong when creating/opening file %s, %s", version, err)
+	}
+
+	hasher := sha256.New()
```

Comment:
We can just simplify **a lot** by not checking hashes and dumbly downloading all the time. 

There's probably not many scenarios where the complexity here justifies the savings? Without migrating to the NVD API, we'd just be running a cron job at some regular interval that's unlikely to hit any rate limiting.

Reply:
Yeah, I originally implemented that because it made local development much faster, but now that it's split up into two programs, I think it's safe to just remove it here. It does take a while to download (~5 minutes or so ) compared to <10 seconds to check, but it isn't long enough to have all this complexity.